### PR TITLE
Add cifar integration test

### PR DESCRIPTION
**Pull request by nvkevlu**
Adds CIFAR-10 as an integration test.

### Description

Adds the job ``cifar10_fedavg_stream_tb`` into integration test cases.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: docs/flare_overview.rst [hunk 1]
```diff
@@ -52,7 +52,7 @@ with a set of federation workflows and learning algorithms.
 Alongside this central stack are tools that allow experimentation and proof-of-concept development
```
Comment (YuanTingHsieh) [inline docs/flare_overview.rst:52]: need to checkout these changes / rebase on dev

File: examples/cifar10/pt/utils/cifar10_data_splitter.py [hunk 2]
```diff
@@ -61,12 +61,13 @@ def _get_site_class_summary(train_label, site_idx):
 
 
 class Cifar10DataSplitter(FLComponent):
-    def __init__(self, split_dir: str = None, num_sites: int = 8, alpha: float = 0.5, seed: int = 0):
+    def __init__(self, split_dir: str = None, num_sites: int = 8, alpha: float = 0.5, seed: int = 0, custom_CIFAR10_ROOT: str = None):
```
Comment (YuanTingHsieh) [inline examples/cifar10/pt/utils/cifar10_data_splitter.py:64]: suggest passing in "cifar_root", or "data_root"

then set "self.cifar_root" or "self.data_root"

File: examples/cifar10/pt/utils/cifar10_data_splitter.py [hunk 3]
```diff
@@ -103,7 +104,10 @@ def split(self, fl_ctx: FLContext):
 
     def load_cifar10_data(self):
         # download data
-        train_dataset = datasets.CIFAR10(root=CIFAR10_ROOT, train=True, download=True)
+        if self.custom_CIFAR10_ROOT:
+            train_dataset = datasets.CIFAR10(root=self.custom_CIFAR10_ROOT, train=True, download=True)
+        else:
+            train_dataset = datasets.CIFAR10(root=CIFAR10_ROOT, train=True, download=True)
```
Comment (YuanTingHsieh) [inline examples/cifar10/pt/utils/cifar10_data_splitter.py:110]: and we can just use same line ``train_dataset = datasets.CIFAR10(root=self.cifar_root, train=True, download=True)``

File: examples/cifar10/pt/utils/cifar10_download_data.py [hunk 4]
```diff
@@ -41,7 +41,7 @@
 
 
 def main():
-    splitter = Cifar10DataSplitter()
+    splitter = Cifar10DataSplitter(custom_CIFAR10_ROOT="/persistenttmp/cifar10")
```
Comment (YuanTingHsieh) [inline examples/cifar10/pt/utils/cifar10_download_data.py:44]: instead of modifying codes here, it would be better if CI can do its own data preparation.

**Comment by nvkevlu** [comment]
Rebased and undid changes on the data splitter

File: tests/integration_test/src/action_handlers.py [hunk 5]
```diff
@@ -142,6 +187,7 @@ def handle(self, command_args: list, admin_controller: FLTestDriver, admin_api:
 class _ListJobHandler(_CmdHandler):
     def handle(self, command_args: list, admin_controller: FLTestDriver, admin_api: FLAdminAPI):
         response = admin_api.list_jobs("-a")
+        print(response)
```
Comment (YuanTingHsieh) [inline tests/integration_test/src/action_handlers.py:190]: do we need the print statement here?

File: tests/integration_test/src/action_handlers.py [hunk 6]
```diff
@@ -110,6 +154,7 @@ def handle(self, command_args: list, admin_controller: FLTestDriver, admin_api:
         job_name = str(command_args[0])
 
         response = admin_api.submit_job(job_name)
+        print(response)
```
Comment (YuanTingHsieh) [inline tests/integration_test/src/action_handlers.py:157]: do we need the print statement here?

File: tests/integration_test/data/test_configs/standalone_job/test_cifar.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+          "data": {}
+      - "trigger":
+          "type": "run_state"
+          "data": { "run_finished": True }
+        "actions": [ "ensure_current_job_done" ]
+        "result":
+          "type": "run_state"
+          "data": { "run_finished": True }
+    validators:
+      - path: tests.integration_test.validators.CIFAR10ResultValidator
```
Comment (YuanTingHsieh) [inline tests/integration_test/data/test_configs/standalone_job/test_cifar.yml:26]: we might need 

```
    setup:
      - python -c "from torchvision.datasets import CIFAR10; CIFAR10(root='~/data', download=True)"
    teardown:
      - rm -rf ~/data
```

For the dataset to be downloaded automatically.
But we need to check what is the PATH that is being used.

File: tests/integration_test/src/action_handlers.py (hunk 5 above)
Comment (nvkevlu) [reply tests/integration_test/src/action_handlers.py:190]: I can remove this

File: tests/integration_test/src/action_handlers.py (hunk 6 above)
Comment (nvkevlu) [reply tests/integration_test/src/action_handlers.py:157]: I can remove this

File: tests/integration_test/data/test_configs/standalone_job/test_cifar.yml (hunk 7 above)
Comment (nvkevlu) [reply tests/integration_test/data/test_configs/standalone_job/test_cifar.yml:26]: Added the data path, but tried this for the export but that did not work for the PYTHONPATH

**Review by YuanTingHsieh — APPROVED** [review]
LGTM as a first pass to add examples into integration tests.
Let's discuss on the shim layer design based on what we found here.

**Pull request merged**